docs/advertisement-layout-management.md

```diff
@@ -0,0 +1,195 @@
+# Advertisement layout management
+
+With ad layout config you can create your own ad break timeline using your vast tags.  
+Ad break can be set as pre, mid and post rolls and each ad break can contain a single vast tag or multiple tags, either as a pod, but also as a [Waterfall](#waterfalling)
+
+## Table of Contents
+
+- [Single Ad](#single-ad)
+- [Ad Pod](#ad-pod)
+- [Waterfalling](#waterfalling)
+- [Ad Options](#ad-options)
+- [Events](#events)
+- [Play Ad Now](#play-ad-now)
+
+### Single Ad
+
+Here's a simple scheme contains 4 ad breaks: 1 pre-roll, 2 mid-rolls and 1 post-roll:
+
+```ecmascript 6
+const config = {
+  ...
+  advertising: {
+    adBreaks: [{
+      position: 0,
+      ads: [{
+          url: [PRE_ROLL_VAST_URL]
+        }]
+      }, {
+      position: 15,
+      ads: [{
+          url: [MID_ROLL_1_VAST_URL]
+        }]
+      }, {
+      position: 60,
+      ads: [{
+          url: [MID_ROLL_2_VAST_URL]
+        }]
+      }, {
+      position: -1,
+      ads: [{
+          url: [POST_ROLL_VAST_URL]
+        }]
+      }
+    ]
+  }
+  ...
+}
+const kalturaPlayer = KalturaPlayer.setup(config);
+```
+
+> In this sample, the player will try to request and play 4 ads.  
+> Note: `position: 0` means pre-roll. `position: -1` means postroll.
+
+### Ad Pod
+
+An ad break may contain a list of ads, also referred to as ad pod.
+
+> Ad Pod definition (from [IAB](https://www.iab.com/insights/digital-video-advertising-glossary)): An individual ad pod is a group of ads expected to play back-to-back in one commercial ad break similar to how consumers experience commercial ad breaks in broadcast television. An ad pod can be of varying lengths and can be inserted at any point in a stream of content (pre, mid, or post).
+
+```ecmascript 6
+const config = {
+  ...
+  advertising: {
+    adBreaks: [{
+      position: 0,
+      ads: [{
+          url: [PRE_ROLL_1_VAST_URL]
+        },{
+          url: [PRE_ROLL_2_VAST_URL]
+        }]
+      }, {
+      position: 15,
+      ads: [{
+          url: [MID_ROLL_1_VAST_URL]
+        },{
+          url: [MID_ROLL_2_VAST_URL]
+        }]
+      }
+    ]
+  }
+  ...
+}
+const kalturaPlayer = KalturaPlayer.setup(config);
+```
+
+> In this sample, the player will try to request and play 4 ads in 2 ad breaks, each one contains 2 ads consecutively.
+
+### Waterfalling
+
+> Waterfalling ("daisy chain") is a process used by publishers selling ads in order to get a better fill-rate.
+> This process can be implemented on the ad server itself, but it can also be implemented on the client - which is what we did.
+> The process is rather simple: if the first ad server doesn't have any ads to serve, a call to a second network is made etc.
+> Meaning, if the first priority server did not "purchase" the impression, it would cascade down to the next server and to the next, until someone bought it.  
```

Comment:
bought? did you brought? maybe replace with "until an success ad is served"?

Reply:
I agree, can be replaced by:

> until ad request is fulfilled, or all ad servers didn't respond with an ad